[2023-05-01T13:25:38.897+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:871: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti: TaskInstance | None = qry.with_for_update().one_or_none()

[2023-05-01T13:25:38.947+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: spark_process_data_12.spark_job_clean_data_12 manual__2023-05-01T13:24:18.650757+00:00 [queued]>
[2023-05-01T13:25:38.983+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: spark_process_data_12.spark_job_clean_data_12 manual__2023-05-01T13:24:18.650757+00:00 [queued]>
[2023-05-01T13:25:38.983+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2023-05-01T13:25:38.984+0000] {taskinstance.py:1363} INFO - Starting attempt 2 of 2
[2023-05-01T13:25:38.984+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2023-05-01T13:25:39.039+0000] {taskinstance.py:1383} INFO - Executing <Task(SparkSubmitOperator): spark_job_clean_data_12> on 2023-05-01 13:24:18.650757+00:00
[2023-05-01T13:25:39.053+0000] {standard_task_runner.py:54} INFO - Started process 294 to run task
[2023-05-01T13:25:39.065+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'spark_process_data_12', 'spark_job_clean_data_12', 'manual__2023-05-01T13:24:18.650757+00:00', '--job-id', '21', '--raw', '--subdir', 'DAGS_FOLDER/spark_process_scheduler.py', '--cfg-path', '/tmp/tmp6z2ib1ca']
[2023-05-01T13:25:39.068+0000] {standard_task_runner.py:83} INFO - Job 21: Subtask spark_job_clean_data_12
[2023-05-01T13:25:39.071+0000] {dagbag.py:525} INFO - Filling up the DagBag from /opt/***/dags/spark_process_scheduler.py
[2023-05-01T13:25:39.372+0000] {task_command.py:384} INFO - Running <TaskInstance: spark_process_data_12.spark_job_clean_data_12 manual__2023-05-01T13:24:18.650757+00:00 [running]> on host 3a81bcbab442
[2023-05-01T13:25:39.382+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-01T13:25:39.661+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=pioneer22022001@gmail.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark_process_data_12
AIRFLOW_CTX_TASK_ID=spark_job_clean_data_12
AIRFLOW_CTX_EXECUTION_DATE=2023-05-01T13:24:18.650757+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-05-01T13:24:18.650757+00:00
[2023-05-01T13:25:39.692+0000] {base.py:71} INFO - Using connection ID 'other_spark_local' for task execution.
[2023-05-01T13:25:39.695+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark_master:7077 --conf spark.master=spark://spark_master:7077 --name spark_clean_data --verbose /opt/spark/app/preprocess_data.py
[2023-05-01T13:25:39.738+0000] {spark_submit.py:490} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2023-05-01T13:25:43.299+0000] {spark_submit.py:490} INFO - Using properties file: null
[2023-05-01T13:25:44.046+0000] {spark_submit.py:490} INFO - Parsed arguments:
[2023-05-01T13:25:44.047+0000] {spark_submit.py:490} INFO - master                  spark://spark_master:7077
[2023-05-01T13:25:44.048+0000] {spark_submit.py:490} INFO - deployMode              null
[2023-05-01T13:25:44.049+0000] {spark_submit.py:490} INFO - executorMemory          null
[2023-05-01T13:25:44.049+0000] {spark_submit.py:490} INFO - executorCores           null
[2023-05-01T13:25:44.050+0000] {spark_submit.py:490} INFO - totalExecutorCores      null
[2023-05-01T13:25:44.050+0000] {spark_submit.py:490} INFO - propertiesFile          null
[2023-05-01T13:25:44.051+0000] {spark_submit.py:490} INFO - driverMemory            null
[2023-05-01T13:25:44.051+0000] {spark_submit.py:490} INFO - driverCores             null
[2023-05-01T13:25:44.052+0000] {spark_submit.py:490} INFO - driverExtraClassPath    null
[2023-05-01T13:25:44.052+0000] {spark_submit.py:490} INFO - driverExtraLibraryPath  null
[2023-05-01T13:25:44.053+0000] {spark_submit.py:490} INFO - driverExtraJavaOptions  null
[2023-05-01T13:25:44.053+0000] {spark_submit.py:490} INFO - supervise               false
[2023-05-01T13:25:44.054+0000] {spark_submit.py:490} INFO - queue                   null
[2023-05-01T13:25:44.054+0000] {spark_submit.py:490} INFO - numExecutors            null
[2023-05-01T13:25:44.055+0000] {spark_submit.py:490} INFO - files                   null
[2023-05-01T13:25:44.055+0000] {spark_submit.py:490} INFO - pyFiles                 null
[2023-05-01T13:25:44.056+0000] {spark_submit.py:490} INFO - archives                null
[2023-05-01T13:25:44.056+0000] {spark_submit.py:490} INFO - mainClass               null
[2023-05-01T13:25:44.057+0000] {spark_submit.py:490} INFO - primaryResource         file:/opt/spark/app/preprocess_data.py
[2023-05-01T13:25:44.057+0000] {spark_submit.py:490} INFO - name                    spark_clean_data
[2023-05-01T13:25:44.058+0000] {spark_submit.py:490} INFO - childArgs               []
[2023-05-01T13:25:44.058+0000] {spark_submit.py:490} INFO - jars                    null
[2023-05-01T13:25:44.059+0000] {spark_submit.py:490} INFO - packages                null
[2023-05-01T13:25:44.059+0000] {spark_submit.py:490} INFO - packagesExclusions      null
[2023-05-01T13:25:44.060+0000] {spark_submit.py:490} INFO - repositories            null
[2023-05-01T13:25:44.060+0000] {spark_submit.py:490} INFO - verbose                 true
[2023-05-01T13:25:44.061+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:44.061+0000] {spark_submit.py:490} INFO - Spark properties used, including those specified through
[2023-05-01T13:25:44.062+0000] {spark_submit.py:490} INFO - --conf and those from the properties file null:
[2023-05-01T13:25:44.062+0000] {spark_submit.py:490} INFO - (spark.master,spark://spark_master:7077)
[2023-05-01T13:25:44.063+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:44.064+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:44.266+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-01T13:25:45.410+0000] {spark_submit.py:490} INFO - Main class:
[2023-05-01T13:25:45.411+0000] {spark_submit.py:490} INFO - org.apache.spark.deploy.PythonRunner
[2023-05-01T13:25:45.412+0000] {spark_submit.py:490} INFO - Arguments:
[2023-05-01T13:25:45.412+0000] {spark_submit.py:490} INFO - file:/opt/spark/app/preprocess_data.py
[2023-05-01T13:25:45.413+0000] {spark_submit.py:490} INFO - null
[2023-05-01T13:25:45.419+0000] {spark_submit.py:490} INFO - Spark config:
[2023-05-01T13:25:45.420+0000] {spark_submit.py:490} INFO - (spark.app.name,spark_clean_data)
[2023-05-01T13:25:45.420+0000] {spark_submit.py:490} INFO - (spark.submit.pyFiles,)
[2023-05-01T13:25:45.421+0000] {spark_submit.py:490} INFO - (spark.submit.deployMode,client)
[2023-05-01T13:25:45.421+0000] {spark_submit.py:490} INFO - (spark.master,spark://spark_master:7077)
[2023-05-01T13:25:45.421+0000] {spark_submit.py:490} INFO - Classpath elements:
[2023-05-01T13:25:45.422+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:45.422+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:45.424+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:47.612+0000] {spark_submit.py:490} INFO - Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[2023-05-01T13:25:47.643+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:47 INFO SparkContext: Running Spark version 3.2.3
[2023-05-01T13:25:48.029+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2023-05-01T13:25:48.391+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceUtils: ==============================================================
[2023-05-01T13:25:48.393+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceUtils: No custom resources configured for spark.driver.
[2023-05-01T13:25:48.394+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceUtils: ==============================================================
[2023-05-01T13:25:48.395+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SparkContext: Submitted application: clean_data
[2023-05-01T13:25:48.480+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2023-05-01T13:25:48.529+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceProfile: Limiting resource is cpu
[2023-05-01T13:25:48.531+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2023-05-01T13:25:48.700+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SecurityManager: Changing view acls to: ***
[2023-05-01T13:25:48.701+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SecurityManager: Changing modify acls to: ***
[2023-05-01T13:25:48.703+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SecurityManager: Changing view acls groups to:
[2023-05-01T13:25:48.704+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SecurityManager: Changing modify acls groups to:
[2023-05-01T13:25:48.706+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:48 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2023-05-01T13:25:49.767+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:49 INFO Utils: Successfully started service 'sparkDriver' on port 39835.
[2023-05-01T13:25:49.862+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:49 INFO SparkEnv: Registering MapOutputTracker
[2023-05-01T13:25:49.971+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:49 INFO SparkEnv: Registering BlockManagerMaster
[2023-05-01T13:25:50.035+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2023-05-01T13:25:50.037+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2023-05-01T13:25:50.048+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2023-05-01T13:25:50.120+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-7fcf8e5e-10a0-4fae-a5c0-c6f62557586a
[2023-05-01T13:25:50.239+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO MemoryStore: MemoryStore started with capacity 366.3 MiB
[2023-05-01T13:25:50.308+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO SparkEnv: Registering OutputCommitCoordinator
[2023-05-01T13:25:50.995+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:50 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2023-05-01T13:25:51.226+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:51 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://3a81bcbab442:4040
[2023-05-01T13:25:51.983+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:51 ERROR SparkContext: Error initializing SparkContext.
[2023-05-01T13:25:51.984+0000] {spark_submit.py:490} INFO - org.apache.spark.SparkException: Invalid master URL: spark://spark_master:7077
[2023-05-01T13:25:51.985+0000] {spark_submit.py:490} INFO - at org.apache.spark.util.Utils$.extractHostPortFromSparkUrl(Utils.scala:2524)
[2023-05-01T13:25:51.985+0000] {spark_submit.py:490} INFO - at org.apache.spark.rpc.RpcAddress$.fromSparkURL(RpcAddress.scala:47)
[2023-05-01T13:25:51.986+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.client.StandaloneAppClient.$anonfun$masterRpcAddresses$1(StandaloneAppClient.scala:53)
[2023-05-01T13:25:51.986+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[2023-05-01T13:25:51.987+0000] {spark_submit.py:490} INFO - at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[2023-05-01T13:25:51.987+0000] {spark_submit.py:490} INFO - at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[2023-05-01T13:25:51.988+0000] {spark_submit.py:490} INFO - at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[2023-05-01T13:25:51.989+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.map(TraversableLike.scala:286)
[2023-05-01T13:25:51.989+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[2023-05-01T13:25:51.990+0000] {spark_submit.py:490} INFO - at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:198)
[2023-05-01T13:25:51.990+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.client.StandaloneAppClient.<init>(StandaloneAppClient.scala:53)
[2023-05-01T13:25:51.991+0000] {spark_submit.py:490} INFO - at org.apache.spark.scheduler.cluster.StandaloneSchedulerBackend.start(StandaloneSchedulerBackend.scala:123)
[2023-05-01T13:25:51.991+0000] {spark_submit.py:490} INFO - at org.apache.spark.scheduler.TaskSchedulerImpl.start(TaskSchedulerImpl.scala:220)
[2023-05-01T13:25:51.992+0000] {spark_submit.py:490} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:581)
[2023-05-01T13:25:51.992+0000] {spark_submit.py:490} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2023-05-01T13:25:51.993+0000] {spark_submit.py:490} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2023-05-01T13:25:51.993+0000] {spark_submit.py:490} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2023-05-01T13:25:51.994+0000] {spark_submit.py:490} INFO - at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2023-05-01T13:25:51.994+0000] {spark_submit.py:490} INFO - at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2023-05-01T13:25:51.995+0000] {spark_submit.py:490} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2023-05-01T13:25:51.995+0000] {spark_submit.py:490} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2023-05-01T13:25:51.996+0000] {spark_submit.py:490} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2023-05-01T13:25:51.996+0000] {spark_submit.py:490} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2023-05-01T13:25:51.997+0000] {spark_submit.py:490} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2023-05-01T13:25:51.997+0000] {spark_submit.py:490} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2023-05-01T13:25:51.998+0000] {spark_submit.py:490} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2023-05-01T13:25:51.998+0000] {spark_submit.py:490} INFO - at java.lang.Thread.run(Thread.java:750)
[2023-05-01T13:25:52.026+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO SparkUI: Stopped Spark web UI at http://3a81bcbab442:4040
[2023-05-01T13:25:52.053+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO StandaloneSchedulerBackend: Shutting down all executors
[2023-05-01T13:25:52.068+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2023-05-01T13:25:52.140+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2023-05-01T13:25:52.166+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO MemoryStore: MemoryStore cleared
[2023-05-01T13:25:52.167+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO BlockManager: BlockManager stopped
[2023-05-01T13:25:52.192+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO BlockManagerMaster: BlockManagerMaster stopped
[2023-05-01T13:25:52.193+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 WARN MetricsSystem: Stopping a MetricsSystem that is not running
[2023-05-01T13:25:52.199+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2023-05-01T13:25:52.220+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO SparkContext: Successfully stopped SparkContext
[2023-05-01T13:25:52.222+0000] {spark_submit.py:490} INFO - Traceback (most recent call last):
[2023-05-01T13:25:52.222+0000] {spark_submit.py:490} INFO - File "/opt/spark/app/preprocess_data.py", line 10, in <module>
[2023-05-01T13:25:52.222+0000] {spark_submit.py:490} INFO - spark = SparkSession.builder.appName('clean_data').getOrCreate()
[2023-05-01T13:25:52.223+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/sql/session.py", line 228, in getOrCreate
[2023-05-01T13:25:52.223+0000] {spark_submit.py:490} INFO - sc = SparkContext.getOrCreate(sparkConf)
[2023-05-01T13:25:52.223+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/context.py", line 392, in getOrCreate
[2023-05-01T13:25:52.224+0000] {spark_submit.py:490} INFO - SparkContext(conf=conf or SparkConf())
[2023-05-01T13:25:52.224+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/context.py", line 147, in __init__
[2023-05-01T13:25:52.225+0000] {spark_submit.py:490} INFO - conf, jsc, profiler_cls)
[2023-05-01T13:25:52.225+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/context.py", line 209, in _do_init
[2023-05-01T13:25:52.225+0000] {spark_submit.py:490} INFO - self._jsc = jsc or self._initialize_context(self._conf._jconf)
[2023-05-01T13:25:52.225+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/context.py", line 329, in _initialize_context
[2023-05-01T13:25:52.226+0000] {spark_submit.py:490} INFO - return self._jvm.JavaSparkContext(jconf)
[2023-05-01T13:25:52.226+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/py4j/java_gateway.py", line 1586, in __call__
[2023-05-01T13:25:52.226+0000] {spark_submit.py:490} INFO - answer, self._gateway_client, None, self._fqn)
[2023-05-01T13:25:52.227+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/py4j/protocol.py", line 328, in get_return_value
[2023-05-01T13:25:52.227+0000] {spark_submit.py:490} INFO - format(target_id, ".", name), value)
[2023-05-01T13:25:52.229+0000] {spark_submit.py:490} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
[2023-05-01T13:25:52.229+0000] {spark_submit.py:490} INFO - : org.apache.spark.SparkException: Invalid master URL: spark://spark_master:7077
[2023-05-01T13:25:52.230+0000] {spark_submit.py:490} INFO - at org.apache.spark.util.Utils$.extractHostPortFromSparkUrl(Utils.scala:2524)
[2023-05-01T13:25:52.231+0000] {spark_submit.py:490} INFO - at org.apache.spark.rpc.RpcAddress$.fromSparkURL(RpcAddress.scala:47)
[2023-05-01T13:25:52.231+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.client.StandaloneAppClient.$anonfun$masterRpcAddresses$1(StandaloneAppClient.scala:53)
[2023-05-01T13:25:52.232+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
[2023-05-01T13:25:52.232+0000] {spark_submit.py:490} INFO - at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[2023-05-01T13:25:52.233+0000] {spark_submit.py:490} INFO - at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[2023-05-01T13:25:52.233+0000] {spark_submit.py:490} INFO - at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[2023-05-01T13:25:52.234+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.map(TraversableLike.scala:286)
[2023-05-01T13:25:52.234+0000] {spark_submit.py:490} INFO - at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
[2023-05-01T13:25:52.235+0000] {spark_submit.py:490} INFO - at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:198)
[2023-05-01T13:25:52.235+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.client.StandaloneAppClient.<init>(StandaloneAppClient.scala:53)
[2023-05-01T13:25:52.236+0000] {spark_submit.py:490} INFO - at org.apache.spark.scheduler.cluster.StandaloneSchedulerBackend.start(StandaloneSchedulerBackend.scala:123)
[2023-05-01T13:25:52.236+0000] {spark_submit.py:490} INFO - at org.apache.spark.scheduler.TaskSchedulerImpl.start(TaskSchedulerImpl.scala:220)
[2023-05-01T13:25:52.237+0000] {spark_submit.py:490} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:581)
[2023-05-01T13:25:52.237+0000] {spark_submit.py:490} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2023-05-01T13:25:52.238+0000] {spark_submit.py:490} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2023-05-01T13:25:52.238+0000] {spark_submit.py:490} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2023-05-01T13:25:52.239+0000] {spark_submit.py:490} INFO - at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2023-05-01T13:25:52.240+0000] {spark_submit.py:490} INFO - at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2023-05-01T13:25:52.240+0000] {spark_submit.py:490} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2023-05-01T13:25:52.241+0000] {spark_submit.py:490} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2023-05-01T13:25:52.241+0000] {spark_submit.py:490} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2023-05-01T13:25:52.242+0000] {spark_submit.py:490} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2023-05-01T13:25:52.242+0000] {spark_submit.py:490} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2023-05-01T13:25:52.243+0000] {spark_submit.py:490} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2023-05-01T13:25:52.243+0000] {spark_submit.py:490} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2023-05-01T13:25:52.244+0000] {spark_submit.py:490} INFO - at java.lang.Thread.run(Thread.java:750)
[2023-05-01T13:25:52.244+0000] {spark_submit.py:490} INFO - 
[2023-05-01T13:25:52.411+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO ShutdownHookManager: Shutdown hook called
[2023-05-01T13:25:52.413+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO ShutdownHookManager: Deleting directory /tmp/spark-a1b6971c-53a3-479a-bafe-e23b984e8eea
[2023-05-01T13:25:52.421+0000] {spark_submit.py:490} INFO - 23/05/01 13:25:52 INFO ShutdownHookManager: Deleting directory /tmp/spark-27758ff2-59da-4187-b447-fc911fe1a937
[2023-05-01T13:25:52.583+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:871: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti: TaskInstance | None = qry.with_for_update().one_or_none()

[2023-05-01T13:25:52.600+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark_master:7077 --conf spark.master=spark://spark_master:7077 --name spark_clean_data --verbose /opt/spark/app/preprocess_data.py. Error code is: 1.
[2023-05-01T13:25:52.606+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-01T13:25:52.617+0000] {taskinstance.py:1406} INFO - Marking task as FAILED. dag_id=spark_process_data_12, task_id=spark_job_clean_data_12, execution_date=20230501T132418, start_date=20230501T132538, end_date=20230501T132552
[2023-05-01T13:25:52.664+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/utils/email.py:120: RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
  send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)

[2023-05-01T13:25:52.665+0000] {configuration.py:569} WARNING - section/key [smtp/smtp_user] not found in config
[2023-05-01T13:25:52.665+0000] {email.py:229} INFO - Email alerting: attempt 1
[2023-05-01T13:25:52.681+0000] {configuration.py:569} WARNING - section/key [smtp/smtp_user] not found in config
[2023-05-01T13:25:52.681+0000] {email.py:229} INFO - Email alerting: attempt 1
[2023-05-01T13:25:52.682+0000] {taskinstance.py:1914} ERROR - Failed to send email to: ['pioneer22022001@gmail.com']
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1457, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1603, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1664, in _execute_task
    result = execute_callable(context=context)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark_master:7077 --conf spark.master=spark://spark_master:7077 --name spark_clean_data --verbose /opt/spark/app/preprocess_data.py. Error code is: 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2311, in email_alert
    send_email(task.email, subject, html_content)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 72, in send_email
    **kwargs,
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 120, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 231, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 265, in _get_smtp_connection
    else smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.7/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 307, in _get_socket
    self.source_address)
  File "/usr/local/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/usr/local/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1912, in handle_failure
    self.email_alert(error, task)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2313, in email_alert
    send_email(task.email, subject, html_content_err)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 72, in send_email
    **kwargs,
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 120, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 231, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 265, in _get_smtp_connection
    else smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.7/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 307, in _get_socket
    self.source_address)
  File "/usr/local/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/usr/local/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
[2023-05-01T13:25:52.728+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 21 for task spark_job_clean_data_12 (Cannot execute: spark-submit --master spark://spark_master:7077 --conf spark.master=spark://spark_master:7077 --name spark_clean_data --verbose /opt/spark/app/preprocess_data.py. Error code is: 1.; 294)
[2023-05-01T13:25:52.755+0000] {local_task_job.py:164} INFO - Task exited with return code 1
[2023-05-01T13:25:52.835+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
